### PR TITLE
Remove currentLocale prepend on iOS

### PR DESCRIPTION
**Pull request by GaryQian**
Partially reverts https://github.com/flutter/engine/pull/6995

Since we handle null country codes now, this prepend should be safe to remove. [NSLocale currentLocale] is a pre-resolved locale, which can result in the first locale being one that is not preferred by the user. See https://github.com/flutter/flutter/issues/49775

### CODE REVIEW

**Comment by GaryQian** [comment]
I don't think our iOS embedder has testing set up yet

**Comment by dnfield** [comment]
There are a few places this should be testable, @gaaclarke and @xster have done some work recently to make it even easier from what I understand

**Comment by dnfield** [comment]
If nothing else, soemthing in the scenarios app should cover this.

**Comment by xster** [comment]
We also have FlutterEngine unit tests

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@end
+
+@implementation XCUIElement (ftr_waitForNonExistence)
+- (BOOL)ftr_waitForNonExistenceWithTimeout:(NSTimeInterval)timeout {
+  NSTimeInterval delta = 0.5;
+  while (timeout > 0.0) {
+    if (!self.exists) {
+      return YES;
+    }
+    usleep(delta * 1000000);
```
Comment (dnfield) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: Can we avoid this? 

Would it help at all if we had EarlyGrey for Flutter instead?

If we need that, we should probably add a TODO here to replace this when EarlGrey is available. 
/cc @jmagman who might know a good trick for this

Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: Something like:
```obj-c
NSPredicate *predicate = [NSPredicate predicateWithFormat:@"exists = YES"];
XCTestExpectation *expectation = [self expectationForPredicate:predicate evaluatedWithObject:textInputSemanticsObject handler:NULL];
[self waitForExpectations:@[expectation] timeout:10.0];
```

Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: This is probably better, actually:
```obj-c
XCTestExpectation *expectation = [self keyValueObservingExpectationForObject:textInputSemanticsObject keyPath:@"exists" expectedValue:@YES];
[self waitForExpectations:@[expectation] timeout:10.0];
```
(I didn't actually test these).

Comment (dnfield) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: It's coming back to me now - that week I got better at Objective C with your guidance .. and how it faded...

Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: I had to look it up myself, it's been awhile...

Comment (GaryQian) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:25]: Ahh thanks for the assistance, Trying to learn all the iOS testing frameworks and whatnot to write this haha. This particular code was repurposed from another test, still working on refining/improving it.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitialization.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Copyright 2020 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <XCTest/XCTest.h>
+
+NS_ASSUME_NONNULL_BEGIN
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitialization.h:8]: Remove the header, you can move the interface into the test .m file.

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- (void)testNoLocalePrepend {
+  NSTimeInterval timeout = 10.0;
+
+  XCUIElement* textInputSemanticsObject =
+      [[[self.application textFields] matchingIdentifier:@"[en]"] element];
+
+  // The locales recieved by dart:ui are exposed onBeginFrame via semantics label.
+  // There should only be one locale, as we have removed the locale prepend on iOS.
+  XCTAssertTrue([textInputSemanticsObject waitForExistenceWithTimeout:timeout]);
+  XCTAssertEqualObjects([textInputSemanticsObject valueForKey:@"hasKeyboardFocus"], @(NO));
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:29]: I'm guessing `textInputSemanticsObject.hasKeyboardFocus` isn't an exposed property?

The better typed way to do this when you really have to (ignoring StackOverflow) is to put this at the top of your implementation file:
```objective-c
@interface XCUIElement (FlutterTests)
@property BOOL hasKeyboardFocus;
@end
...

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--locale-initialization" ];
+  [self.application launch];
+}
+
+- (void)testNoLocalePrepend {
+  NSTimeInterval timeout = 10.0;
+
+  XCUIElement* textInputSemanticsObject =
```
Comment (jmagman) [inline testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:23]: I know the engine has variable usage of dot notation, but the "modern" syntax (like circa 2006 modern) of properties would be:
```objective-c
XCUIElement* textInputSemanticsObject = [self.application.textFields matchingIdentifier:@"[en]"].element;
```

File: testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m (hunk 3 above)
Comment (jmagman) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:29]: And then:
```objective-c
XCTAssertFalse(textInputSemanticsObject.hasKeyboardFocus);
```

Comment (GaryQian) [reply testing/scenario_app/ios/Scenarios/ScenariosUITests/LocalizationInitializationTest.m:29]: Ok, will keep in mind. I will actually remove this here as it is exercised in other tests.

File: shell/platform/darwin/ios/framework/Source/FlutterEngine.mm [hunk 5]
```diff
@@ -762,22 +762,6 @@ - (void)onLocaleUpdated:(NSNotification*)notification {
   NSArray<NSString*>* preferredLocales = [NSLocale preferredLanguages];
   NSMutableArray<NSString*>* data = [[NSMutableArray new] autorelease];
 
-  // Force prepend the [NSLocale currentLocale] to the front of the list
```
Comment (jmagman) [inline shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:765]: Fun fact: you can change the app's locale and language with launch args `-AppleLanguages` and  `-AppleLocale`.
https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/TestingYourInternationalApp/TestingYourInternationalApp.html#//apple_ref/doc/uid/10000171i-CH7-SW2

> Alternatively, add AppleLanguages and AppleLocale launch arguments using the scheme editor—for example, add -AppleLanguages "(de)" to specify the German language and -AppleLocale "fr_FR" to specify the France region.

Comment (GaryQian) [reply shell/platform/darwin/ios/framework/Source/FlutterEngine.mm:765]: Thanks, Will be super useful for the next set of changes that I'll be making immediately after this one!

**Comment by GaryQian** [comment]
Mac iOS Engine LUCI passes on rerun: https://ci.chromium.org/p/flutter/builders/try/Mac%20iOS%20Engine/2826

**Pull request merged**